internal/common/states.go

```diff
@@ -0,0 +1,79 @@
+package common
```

Comment:
For this and the similar commits later on: Please explain in the commit message how (if?) this differs from the original code, if there are any behavioural changes.

Reply:
ok, done